x-pack/plugins/siem/server/graphql/hosts/resolvers.ts

```diff
@@ -24,12 +37,28 @@ export const createHostsResolvers = (
 ): {
   Source: {
     Hosts: QueryHostsResolver;
+    HostDetails: QueryHostDetailsResolver;
+    HostFirstLastSeen: QueryHostFirstLastSeenResolver;
   };
 } => ({
   Source: {
     async Hosts(source, args, { req }, info) {
       const options = createOptions(source, args, info);
       return libs.hosts.getHosts(req, options);
     },
+    async HostDetails(source, args, { req }, info) {
+      const fields = getFields(getOr([], 'fieldNodes[0]', info));
+      const options = {
+        sourceConfiguration: source.configuration,
+        fields: fields.map(field => field.replace('edges.node.', '')),
```

Comment:
Any specific reason you used this instead of [createOptions](https://github.com/elastic/kibana/blob/feature-secops/x-pack/plugins/siem/server/utils/build_query/create_options.ts#L30)?

Reply:
Yes, because I am using `filterquery` to get host detail